alphacsc/tests/test_update_d_multi.py

```diff
@@ -5,81 +5,13 @@
 from alphacsc.loss_and_gradient import compute_objective
 from alphacsc.loss_and_gradient import gradient_d, gradient_uv
 from alphacsc.update_d_multi import _get_d_update_constants
-from alphacsc.update_d_multi import check_solver_and_constraints
 from alphacsc.utils.whitening import whitening
 from alphacsc.utils import construct_X_multi
 
 
 DEBUG = True
 
 
-@pytest.mark.parametrize('solver_d', ['auto', 'fista'])
-@pytest.mark.parametrize('uv_constraint', ['auto'])
-def test_check_solver_and_constraints(solver_d, uv_constraint):
-    """Tests for valid values when rank1 is False."""
-
-    solver_d_, uv_constraint_ = check_solver_and_constraints(False, solver_d,
-                                                             uv_constraint)
-
-    assert solver_d_ == 'fista'
-    assert uv_constraint_ == 'auto'
-
-
-@pytest.mark.parametrize('solver_d', ['auto', 'fista'])
-@pytest.mark.parametrize('uv_constraint', ['joint', 'separate'])
-def test_check_solver_and_constraints_error(solver_d, uv_constraint):
-    """Tests for the case rank1 is False and params are not compatible."""
-
-    with pytest.raises(AssertionError,
-                       match="If rank1 is False, uv_constraint should be*"):
-
-        check_solver_and_constraints(False, solver_d, uv_constraint)
-
-
-@pytest.mark.parametrize('solver_d', ['auto', 'alternate',
-                                      'alternate_adaptive'])
-@pytest.mark.parametrize('uv_constraint', ['auto', 'separate'])
-def test_check_solver_and_constraints_rank1_alternate(solver_d, uv_constraint):
-    """Tests for valid values when solver is alternate and rank1 is True."""
-
-    solver_d_, uv_constraint_ = check_solver_and_constraints(True, solver_d,
-                                                             uv_constraint)
-
-    if solver_d == 'auto':
-        solver_d = 'alternate_adaptive'
-
-    assert solver_d_ == solver_d
-    assert uv_constraint_ == 'separate'
-
-
-@pytest.mark.parametrize('solver_d', ['joint', 'fista'])
-@pytest.mark.parametrize('uv_constraint', ['auto', 'joint', 'separate'])
-def test_check_solver_and_constraints_rank1(solver_d, uv_constraint):
-    """Tests for valid values when solver_d is either in 'joint' or 'fista' and
-    rank1 is True."""
-
-    solver_d_, uv_constraint_ = check_solver_and_constraints(True, solver_d,
-                                                             uv_constraint)
-
-    if uv_constraint == 'auto':
-        uv_constraint = 'joint'
-
-    assert solver_d_ == solver_d
-    assert uv_constraint_ == uv_constraint
-
```

Comment:
Why did you removed these tests?

Reply:
I moved the function to `DSolver` and tests to `test_d_solver.py`

Reply:
ok, sorry for the noise, I did not notice.